src-tauri/src/widget_api/fs/apis.rs

```diff
@@ -0,0 +1,154 @@
+use crate::states::WidgetBaseDirectoryState;
+use std::io::Write;
+use std::path::PathBuf;
+use tauri::{command, AppHandle, Manager, Runtime};
+
+fn get_file_path<R: Runtime>(
+    app_handle: &AppHandle<R>,
+    widget_id: &str,
+    path: &str,
+) -> PathBuf {
+    let widget_base = &app_handle.state::<WidgetBaseDirectoryState>().0;
+    // if storage directory does not exist, create it
+    let widget_dir = widget_base.join(widget_id).join("storage");
+    if !widget_dir.exists() {
+        std::fs::create_dir_all(&widget_dir)
+            .expect("Failed to create widget storage directory");
```

Comment:
```suggestion
        std::fs::create_dir_all(&widget_dir)?;
```

Do you really mean to panic here?

Reply:
Will fix this in the next commit